sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/MethodOverloadsShouldBeGroupedBase.cs

```diff
@@ -24,99 +24,111 @@
+            var misplacedOverloads = new Dictionary<MemberInfo, List<MemberInfo>>();
+            MemberInfo previous = null;
             foreach (var member in members)
             {
-                if (GetMethodName(member, IsCaseSensitive) is string methodName
-                    && IsValidMemberForOverload(member))
+                if (CreateMemberInfo(c, member) is MemberInfo current)
                 {
-                    // Groups that should be together are defined by accessibility, abstract, static and member name #4136
-                    var accessibility = c.SemanticModel.GetDeclaredSymbol(member)?.DeclaredAccessibility.ToString();
-                    key = $"{accessibility}/{IsStatic(member)}/{IsAbstract(member)}/{methodName}";
-                    if (misplacedOverloads.TryGetValue(key, out var currentList))
+                    if (misplacedOverloads.TryGetValue(current, out var values))
                     {
-                        if (!key.Equals(previousKey, CaseSensitivity))
+                        if (!current.Equals(previous))
                         {
-                            currentList.Add(member);
+                            values.Add(current);
                         }
                     }
                     else
                     {
-                        misplacedOverloads.Add(key, new List<TMemberDeclarationSyntax> { member });
+                        misplacedOverloads.Add(current, new List<MemberInfo> { current });
                     }
-                    previousKey = key;
+                    previous = current;
                 }
                 else
                 {
-                    previousKey = null;
+                    previous = null;
                 }
             }
-            return misplacedOverloads;
+            return misplacedOverloads.Values.Where(x => x.Count > 1).ToArray();
         }
 
-        private string GetMethodName(TMemberDeclarationSyntax member, bool caseSensitive)
+        protected class MemberInfo
         {
-            var nameSyntaxNode = GetNameSyntaxNode(member);
-            if (caseSensitive)
+
+            public readonly TMemberDeclarationSyntax Member;
+            public readonly string Accessibility;
+            public readonly SyntaxToken NameSyntax;
+            public readonly bool IsStatic;
+            public readonly bool IsAbstract;
+
+            private readonly bool isCaseSensitive;
+
+            public MemberInfo(SyntaxNodeAnalysisContext context, TMemberDeclarationSyntax member, SyntaxToken nameSyntax, bool isStatic, bool isAbstract, bool isCaseSensitive)
+            {
+                Member = member;
+                Accessibility = context.SemanticModel.GetDeclaredSymbol(member)?.DeclaredAccessibility.ToString();
+                NameSyntax = nameSyntax;
+                IsStatic = isStatic;
+                IsAbstract = isAbstract;
+                this.isCaseSensitive = isCaseSensitive;                
```

Comment:
_minor_ trailing spaces here (there's a VS pluging Whitespace Visualizer that helps with that)